src/main/java/org/opensearch/knn/index/KNNSettings.java

```diff
@@ -499,6 +512,22 @@ public static boolean isFaissAVX2Disabled() {
         }
     }
 
+    public static boolean isFaissAVX512Disabled() {
+        try {
+            return KNNSettings.state().getSettingValue(KNNSettings.KNN_FAISS_AVX512_DISABLED);
```

Comment:
Can we do proper null checks here? In general, I think its best to avoid catching all exceptions.

Reply:
Looks like java boolean cannot be null. So null check won't be possible. 
Your second point on exceptions is valid, and this code shouldn't throw exceptions as a default value is set. I've removed the try/catch block.

Reply:
@akashsha1 as users will manually set this setting in opensearch.yml, there is a chance of getting null. To avoid it shall we change it to 
`return Booleans.parseBoolean(KNNSettings.state().getSettingValue(KNNSettings.KNN_FAISS_AVX512_DISABLED).toString(), false);`

Reply:
`Booleans.parseBoolean` will do the null validation and if it is null, it will return the default value

Reply:
spoke with Naveen on slack, and updated to 
return Booleans.parseBoolean(KNNSettings.state().getSettingValue(KNNSettings.KNN_FAISS_AVX512_DISABLED).toString(), KNN_DEFAULT_FAISS_AVX512_DISABLED_VALUE);